pkg/controller/controller_instance.go

```diff
@@ -1093,10 +1093,14 @@ func setServiceInstanceConditionInternal(toUpdate *v1beta1.ServiceInstance,
 func (c *controller) updateServiceInstanceReferences(toUpdate *v1beta1.ServiceInstance) (*v1beta1.ServiceInstance, error) {
 	pcb := pretty.NewContextBuilder(pretty.ServiceInstance, toUpdate.Namespace, toUpdate.Name)
 	glog.V(4).Info(pcb.Message("Updating references"))
+	status := toUpdate.Status
 	updatedInstance, err := c.serviceCatalogClient.ServiceInstances(toUpdate.Namespace).UpdateReferences(toUpdate)
 	if err != nil {
 		glog.Errorf(pcb.Messagef("Failed to update references: %v", err))
 	}
+	// The UpdateReferences method ignores status changes.
+	// Restore status that might have changed locally to be able to update it later.
+	updatedInstance.Status = status
```

Comment:
Instead of moving the `ObservedGeneration` change until after `updateServiceInstanceReferences()` method is invoked, I just "fixed" the updateServiceInstanceReferences method so that ordering doesn't matter anymore.
I think it's better than having to remember that the ordering matters.

Reply:
I like it!

Reply:
Yup, seems reasonable, and with the no-multi-update changes coming it probably won't end up mattering soon anyway. :smiley: